pkg/virt-handler/vm.go

```diff
@@ -2655,6 +2660,9 @@ func (d *VirtualMachineController) vmUpdateHelperDefault(origVMI *v1.VirtualMach
 		if err != nil {
 			return fmt.Errorf("failed to adjust resources: %v", err)
 		}
+		if err := daemons.MountDaemonsSockets(vmi); err != nil {
```

Comment:
I wonder why you need to mount the socket in 'hotplug' fashion. If the daemon can create the socket before virt-launcher is started, then you can request it when rendering the pod manifest.

Reply:
@vasiliy-ul it is because I want to use a single pr-helper-daemon per node. This implies that we share the same pr-helper socket between all the VMs on the same node. What you propose is creating a pr-helper daemon per VM. This is complex because the pr-helper daemon requires privileged= true and I want to run it in the kubevirt namespace inside the virt-handler pod.

Reply:
We could create a pr-helper privileged pod per VM in a privileged namespace (like kubevirt). But if we have 5000 VM requesting persistent reservation in the cluster, then we'll ended-up with 5000 pr-helper pods. We also evaluate creating a pr-helper process per VM in a pod where we have a monitoring process, but not sure if this is really container and it will be quite hard to estimate the resource to be assigned to that pod

Reply:
Hm... Maybe I am missing something obvious, but even with a single daemon per node, you can mount the socket to multiple virt-launcher's when rendering the pod manifest. E.g. by adding `hostPath` volume mount to the virt-launcher pod definition:  

```yaml
      volumes:
      - name: pr-sock
        hostPath:
          path: /path/on/the/node/pr.sock
          type: Socket
```

Reply:
I would discourage the usage of hostPath as it goes against K8S security.

Reply:
Please correct me, but I think hostPath are restricted to privileged pod, and virt-launcher isn't

Reply:
> I would discourage the usage of hostPath as it goes against K8S security.

Does it also apply to host path type `Socket`?

> Please correct me, but I think hostPath are restricted to privileged pods, and virt-launcher isn't

Good point, honestly I haven't checked that :) According to the docs:

> The files or directories created on the underlying hosts are only writable by root. You either need to run your process as root in a [privileged Container](https://kubernetes.io/docs/tasks/configure-pod-container/security-context/) or modify the file permissions on the host to be able to write to a hostPath volume

If I understand correctly, chown'ing the socket to 107:107 should allow unprivileged access.

Reply:
More the hotplugged volumes, you can think about this as container disks that want to share the same underlying base image

Reply:
I can try with the Socket type if it complains. If it works, it would simplify my life quite a bit :)